contrib/kafka/filters/network/source/mesh/upstream_kafka_client.h

```diff
@@ -17,6 +17,13 @@ namespace Mesh {
 // in case of success this means offset (if acks > 0), or error code.
 struct DeliveryMemento {
 
+  // Pointer to byte array that was passed to Kafka producer.
+  // We use this to tell apart messages.
+  // Important: we do not free this memory, it's still part of the 'ProduceRequestHandler' object.
+  // Future work: adopt Kafka's opaque-pointer functionality so we use less memory instead of
+  // keeping whole payload until we receive a confirmation.
```

Comment:
[future work] we'd be sending [with this](https://github.com/edenhill/librdkafka/blob/v1.7.0/src-cpp/rdkafkacpp.h#L3137) and getting back the same pointer in [callback's argument](https://github.com/edenhill/librdkafka/blob/v1.7.0/src-cpp/rdkafkacpp.h#L2309)